SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -40,6 +40,14 @@ FLocalDeploymentManager::FLocalDeploymentManager()
 {
 }
 
+FLocalDeploymentManager::~FLocalDeploymentManager()
+{
+	// The idea behind this lock is to try and make the thread calling the destructor wait for the thread that is cleaning up processes
+	// if something tries to run on this object after the destructor then we're ruined anyway
+	StoppingDeployment.Lock();
```

Comment:
Is there no guard available for a critical section? Maybe its worth writing one if non is available, all this manual lock / unlock pairing and the explicit try / catch handling makes me nervous.

Reply:
Haha, I had a similar reaction.
Maybe there is something, but I haven't looked into the Unreal threading stuff too much.
@MCArth-improbable if you want to pursue this further.

Reply:
Like @mironec mentioned we may be better off using a better threading primitive.

Reply:
I don't have a lot of experience with unreals mutexes and the like but it seems like `FScopeLock` might be what you want.

Reply:
Ooooh I can make construct FScopeLocks with my critical section, you're right that's definitely better

Reply:
FScopeLock is much preferable yes.

There is also TScopeTryLock, and some others.

Reply:
Note FScopeLock should be used minimally in-code, i.e. only wrap code which absolutely requires a lock, not an entire function body if it can be avoided.

Reply:
I just pushed a commit with this, but I'm not creating the locks right at the start of the `trystopdeploymentgracefully` block. I could create another scope so it is right at the start - would that be better or is what I have fine?